tm2/pkg/bft/rpc/lib/server/handlers.go

```diff
@@ -332,130 +333,59 @@ func makeHTTPHandler(rpcFunc *RPCFunc, logger *slog.Logger) http.HandlerFunc {
-			values[i] = v
-			continue
-		}
+		// Handle hex string
+		if strings.HasPrefix(arg, "0x") {
+			decoded, err := hex.DecodeString(arg[2:])
+			if err != nil {
+				return nil, errors.Wrap(err, "error decoding hex string")
+			}
 
-		values[i], err = jsonStringToArg(argType, arg)
-		if err != nil {
-			return nil, err
-		}
-	}
+			data, err := amino.MarshalJSON(decoded)
+			if err != nil {
+				return nil, errors.Wrap(err, "error marshaling argument to JSON")
+			}
 
-	return values, nil
-}
+			paramsMap[argName] = data
 
-func jsonStringToArg(rt reflect.Type, arg string) (reflect.Value, error) {
-	rv := reflect.New(rt)
-	err := amino.UnmarshalJSON([]byte(arg), rv.Interface())
-	if err != nil {
-		return rv, err
-	}
-	rv = rv.Elem()
-	return rv, nil
-}
-
-func nonJSONStringToArg(rt reflect.Type, arg string) (reflect.Value, error, bool) {
-	if rt.Kind() == reflect.Ptr {
-		rv_, err, ok := nonJSONStringToArg(rt.Elem(), arg)
-		switch {
-		case err != nil:
-			return reflect.Value{}, err, false
-		case ok:
-			rv := reflect.New(rt.Elem())
-			rv.Elem().Set(rv_)
-			return rv, nil, true
-		default:
-			return reflect.Value{}, nil, false
+			continue
 		}
-	} else {
-		return _nonJSONStringToArg(rt, arg)
-	}
-}
 
-var reInt = regexp.MustCompile(`^-?[0-9]+$`)
-
-// NOTE: rt.Kind() isn't a pointer.
-func _nonJSONStringToArg(rt reflect.Type, arg string) (reflect.Value, error, bool) {
-	isIntString := reInt.Match([]byte(arg))
-	isQuotedString := strings.HasPrefix(arg, `"`) && strings.HasSuffix(arg, `"`)
-	isHexString := strings.HasPrefix(strings.ToLower(arg), "0x")
-
-	var expectingString, expectingByteSlice, expectingInt bool
-	switch rt.Kind() {
-	case reflect.Int, reflect.Uint, reflect.Int8, reflect.Uint8, reflect.Int16, reflect.Uint16, reflect.Int32, reflect.Uint32, reflect.Int64, reflect.Uint64:
-		expectingInt = true
-	case reflect.String:
-		expectingString = true
-	case reflect.Slice:
-		expectingByteSlice = rt.Elem().Kind() == reflect.Uint8
-	}
-
-	if isIntString && expectingInt {
-		qarg := `"` + arg + `"`
-		// jsonStringToArg
-		rv, err := jsonStringToArg(rt, qarg)
-		if err != nil {
-			return rv, err, false
-		}
+		// Handle base64 string
+		if decoded, err := base64.StdEncoding.DecodeString(arg); err == nil {
+			data, err := amino.MarshalJSON(decoded)
+			if err != nil {
+				return nil, errors.Wrap(err, "error marshaling argument to JSON")
+			}
 
-		return rv, nil, true
-	}
+			paramsMap[argName] = data
 
-	if isHexString {
-		if !expectingString && !expectingByteSlice {
-			err := errors.New("got a hex string arg, but expected '%s'",
-				rt.Kind().String())
-			return reflect.ValueOf(nil), err, false
+			continue
 		}
 
-		var value []byte
-		value, err := hex.DecodeString(arg[2:])
-		if err != nil {
-			return reflect.ValueOf(nil), err, false
-		}
-		if rt.Kind() == reflect.String {
-			return reflect.ValueOf(string(value)), nil, true
+		// Handle int string
+		if reInt.Match([]byte(arg)) {
+			arg = fmt.Sprintf("%q", arg)
```

Comment:
No, I only quote in the case of integers. You need to format it as a JSON string by quoting it; otherwise, you will encounter an error when calling UnmarshalJSON